app/src/main/java/io/intrepid/thirdpartylibraryexcercise/MainActivity.java

```diff
@@ -62,17 +67,20 @@ public void dismissLoadingIndicator() {
 
     @Override
     public void loadImageFromUrl(String url) {
-        DownloadCatImageTask downloadCatImageTask = new DownloadCatImageTask();
-        downloadCatImageTask.setCallback(this);
-        downloadCatImageTask.execute(url);
-    }
+        Picasso.with(this).load(url).into(imageView, new com.squareup.picasso.Callback() {
+            @Override
+            public void onSuccess() {
+                dismissLoadingIndicator();
+            }
 
-    @Override
-    public void onImageResult(Bitmap bitmap) {
-        dismissLoadingIndicator();
-        imageView.setImageBitmap(bitmap);
+            @Override
+            public void onError() {
+
```

Comment:
we probably want to dismiss the loading indicator `onError` too, since if it actually errors, the user will be stuck with a loading spinner. Good job on adding the picasso callback though!